File: limbo/library-support/src/main/java/co/windly/limbo/activity/fragment/LimboFragmentActivityView.java
package co.windly.limbo.activity.fragment;

import android.support.annotation.NonNull;
import co.windly.limbo.activity.base.LimboActivityView;
import co.windly.limbo.fragment.base.LimboFragmentView;
import me.yokeyword.fragmentation.ISupportActivity;

public interface LimboFragmentActivityView extends LimboActivityView, ISupportActivity {

  //region Fragment

  void loadRootFragment(int containerId, @NonNull LimboFragmentView toFragment);

  void start(LimboFragmentView toFragment);

  void start(LimboFragmentView toFragment, @LimboFragmentView.LaunchMode int launchMode);

  void startWithPopTo(LimboFragmentView toFragment, Class<?> targetFragmentClass, boolean includeTargetFragment);

  void pop();

  void popTo(Class<?> targetFragmentClass, boolean includeTargetFragment);

  void popTo(Class<?> targetFragmentClass, boolean includeTargetFragment, Runnable afterPopTransactionRunnable);

  void popTo(
    Class<?> targetFragmentClass, boolean includeTargetFragment, Runnable afterPopTransactionRunnable, int popAnim);

  LimboFragmentView getTopFragment();

  <F extends LimboFragmentView> F findFragment(Class<F> fragmentClass);

  //endregion
}

File: limbo/library/src/main/java/co/windly/limbo/activity/fragment/LimboFragmentActivityView.kt
package co.windly.limbo.activity.fragment

import co.windly.limbo.LimboView
import co.windly.limbo.fragment.base.LimboFragmentView
import me.yokeyword.fragmentation.ISupportActivity
import me.yokeyword.fragmentation.ISupportFragment.LaunchMode

interface LimboFragmentActivityView : LimboView, ISupportActivity {

  //region Fragment

  val topFragment: LimboFragmentView

  fun loadRootFragment(containerId: Int, toFragment: LimboFragmentView)

  fun start(toFragment: LimboFragmentView)

  fun start(toFragment: LimboFragmentView, @LaunchMode launchMode: Int)

  fun startWithPopTo(toFragment: LimboFragmentView, targetFragmentClass: Class<*>, includeTargetFragment: Boolean)

  fun pop()

  fun popTo(targetFragmentClass: Class<*>, includeTargetFragment: Boolean)

  fun popTo(targetFragmentClass: Class<*>, includeTargetFragment: Boolean, afterPopTransactionRunnable: Runnable)

  fun popTo(
    targetFragmentClass: Class<*>, includeTargetFragment: Boolean, afterPopTransactionRunnable: Runnable, popAnim: Int)

  fun <F : LimboFragmentView> findFragment(fragmentClass: Class<F>): F?

  //endregion
}
